thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -140,21 +140,46 @@ class CDistributorBase : implements IHashDistributor, implements IExceptionHandl
              */
             dedupList.sort(*iCompare, 1);
 
-            OwnedConstThorRow prev;
-            for (unsigned i = c; i>0;)
+            if (!keepBestCompare)
             {
-                OwnedConstThorRow row = dedupList.getClear(--i);
-                if ((NULL != prev.get()) && (0 == iCompare->docompare(prev, row)))
+                OwnedConstThorRow prev;
+                for (unsigned i = c; i>0;)
                 {
-                    /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
-                     * which will be deducted when sent.
-                     */
+                    OwnedConstThorRow row = dedupList.getClear(--i);
+                    if ((NULL != prev.get()) && (0 == iCompare->docompare(prev, row)))
+                    {
+                        /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
+                        * which will be deducted when sent.
+                        */
+                    }
+                    else
+                    {
+                        prev.set(row);
+                        rows.enqueue(row.getClear());
+                    }
                 }
-                else
+            }
+            else
+            {
+                // Queue all the "best" rows
+                OwnedConstThorRow prev;
+                unsigned i = c;
+                prev.set(dedupList.getClear(--i));
+                while (i>0)
                 {
-                    prev.set(row);
-                    rows.enqueue(row.getClear());
+                    OwnedConstThorRow row = dedupList.getClear(--i);
+
+                    if ((NULL != prev.get()) && (0 == iCompare->docompare(prev, row)))
+                    {
+                        // Skip rows that are not "better" rows
+                        // N.B. queue rows that equal in terms of the "best" condition as slave activity select which to retain
+                        if (keepBestCompare->docompare(prev,row) < 0)
+                            continue;
+                    }
+                    rows.enqueue(prev.getClear());
```

Comment:
MORE: Worth talking to @jakesmith about this change.  This code should be in the else clause.  You will not get wrong results, but it will not be filtering very efficiently.  Jake - is it still worth doing this?  Is my understanding correct...